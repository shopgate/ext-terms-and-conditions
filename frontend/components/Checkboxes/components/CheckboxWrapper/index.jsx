import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from './components/Checkbox';

/**
 * Product Specific checkbox wrapper
 * @param {Object} props props
 * @returns {JSX}
 */
const CheckboxWrapper = (props) => {
  const checkboxes = props.product.productCheckboxValues.map((value, index) => (
    <Checkbox
      wrapperIndex={props.wrapperIndex}
      key={index.toString()}
      value={index}
      label={value.text}
      checked={props.checkedValues[index].checked}
      productOnChange={props.handleClick}
      textColor={value.textColor}
    />
  ));
  return checkboxes;
};

CheckboxWrapper.propTypes = {
  wrapperIndex: PropTypes.number.isRequired,
  checkedValues: PropTypes.arrayOf(PropTypes.shape()),
  handleClick: PropTypes.func,
  product: PropTypes.shape(),
};

CheckboxWrapper.defaultProps = {
  checkedValues: [],
  handleClick: () => {},
  product: {},
};

export default CheckboxWrapper;
