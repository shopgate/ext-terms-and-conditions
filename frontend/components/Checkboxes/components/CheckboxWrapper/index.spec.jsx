import React from 'react';
import { mount } from 'enzyme';
import { createMockStore } from '@shopgate/pwa-common/store';
import { Provider } from 'react-redux';
import CheckboxWrapper from './index';

jest.mock('./components/Checkbox', () => () => <div>Mocked Checkboxes</div>);

const store = createMockStore();
const mockedProduct = {
  productCheckboxValues: [
    {
      checked: false,
    },
    {
      checked: false,
    },
    {
      checked: false,
    },
  ],
};
const mockedCheckedValues = [
  {
    checked: false,
  },
  {
    checked: false,
  },
  {
    checked: false,
  },
];
describe('CheckboxWrapper', () => {
  it('should return null if empty checkedValues passed', () => {
    const component = mount((
      <Provider store={store}>
        <CheckboxWrapper
          wrapperIndex={0}
          product={mockedProduct}
          checkedValues={[]}
        />
      </Provider>
    ));
    expect(component.html()).toBe('');
  });
  it('should render', () => {
    const component = mount((
      <Provider store={store}>
        <CheckboxWrapper
          wrapperIndex={0}
          product={mockedProduct}
          checkedValues={mockedCheckedValues}
        />
      </Provider>
    ));
    expect(component.html()).not.toBe(null);
    expect(component).toMatchSnapshot();
  });
});
