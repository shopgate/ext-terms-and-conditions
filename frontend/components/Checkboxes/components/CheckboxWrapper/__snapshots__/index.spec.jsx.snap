// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWrapper should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CheckboxWrapper
    checkedValues={
      Array [
        Object {
          "checked": false,
        },
        Object {
          "checked": false,
        },
        Object {
          "checked": false,
        },
      ]
    }
    handleClick={[Function]}
    product={
      Object {
        "productCheckboxValues": Array [
          Object {
            "checked": false,
          },
          Object {
            "checked": false,
          },
          Object {
            "checked": false,
          },
        ],
      }
    }
    wrapperIndex={0}
  >
    <Component
      checked={false}
      key="0"
      onChange={[Function]}
      value={0}
      wrapperIndex={0}
    >
      <div>
        Mocked Checkboxes
      </div>
    </Component>
    <Component
      checked={false}
      key="1"
      onChange={[Function]}
      value={1}
      wrapperIndex={0}
    >
      <div>
        Mocked Checkboxes
      </div>
    </Component>
    <Component
      checked={false}
      key="2"
      onChange={[Function]}
      value={2}
      wrapperIndex={0}
    >
      <div>
        Mocked Checkboxes
      </div>
    </Component>
  </CheckboxWrapper>
</Provider>
`;
